main.tf

```diff
@@ -74,7 +74,7 @@ resource "oci_core_instance" "this" {
   }
 
   create_vnic_details {
-    assign_public_ip = var.assign_public_ip
+    assign_public_ip = var.public_ip == null ? var.assign_public_ip : false
```

Comment:
Please assign a default value - don't leave to null

Reply:
please also rename the instance from `this` to something more meaningful.

Reply:
This test is wrong and needs to be updated. Thank you for catching it.
It always equal to `false` as `null`is not a valid entry for `var.public_ip`

Will do the renaming

Reply:
I'm a bit confused about this condition check and the variable. Your var.public_ip description says:

"OCID of the Public IP to attach to primary vnic. Valid values are NONE, RESERVED or EPHEMERAL." 

The `assign_public_ip` is a boolean. So, you can't technically put an OCID here. At most, you can say whether this vnic can be assigned a public or private ip. I see you are doing all the other things for reserved public IP etc. 

Please clarify

Reply:
`var.public_ip` is only used for comparison, its value is never assigned to `assign_public_ip` argument.
The test is here for backward compatibility until v3.0.0, to allow using `assign_public_ip` argument instead of the new feature which use `oci_core_public_ip` as a dedicated resource, outside of the `create_vnic_details` block.

The new feature is dropping usage of `assign_public_ip` in the `create_vnic_details` block in favor of a dedicated resource. This is because `create_vnic_details` block allows only for ephemeral public IP.

So the logic is the following:

```
if the new feature is not used (`var.public_ip` has not been initialized -> value is `null`):
    then use the old public ip assignement method with `var.assign_public_ip`
Else
   `var.public_ip` has been initialized, this means the user wants to use the new feature
    disable `assign_public_ip` by setting it to `false`
```

That being said, `var.public_ip` definition needs to be updated, as the current validation condition do not allow `null` and the default is `"none"`.